src/main/java/org/example/exercises/exerciseNo2/NumberInputStatistics.java

```diff
@@ -0,0 +1,61 @@
+package org.example.exercises.exerciseNo2;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class NumberInputStatistics {
+    private static final ArrayList <Integer> numbers= new ArrayList<>();
+
+    public static void main(String[] args) {
+        readInputData();
+        calculateDeciles();
+        calculateStatistics();
+        System.out.println(numbers);
+    }
+
+    private static void readInputData() {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        int n;
+        do {
+            System.out.print("Enter a number between 1 and 1000,to exit print 0: ");
+            try {
+                n = Integer.parseInt(br.readLine());
+                if (n < 1 || n > 1000) {
+                    System.out.println("Invalid input. Enter a number between 1 and 1000.");
```

Comment:
this catches 0 too and when You can exit loop then you need to write unparsable number thos don't corelate to Your message.